src/states/node.rs

```diff
@@ -986,6 +996,22 @@ impl Node {
         let start = Instant::now();
         let rp_object = ResourceProof::new(target_size, difficulty);
         let mut proof = rp_object.create_proof_data(&seed);
+        let leading_zero_bytes = rp_object.create_proof(&mut proof);
+        let parts = proof.into_iter()
+            .chunks(MAX_PART_LEN)
+            .into_iter()
+            .map(|chunk| chunk.collect_vec())
+            .collect_vec();
+        let part_count = parts.len();
+        for (part_index, part) in parts.into_iter().enumerate() {
+            let direct_message = DirectMessage::ResourceProofResponse {
+                part_index: part_index,
+                part_count: part_count,
+                proof: part,
+                leading_zero_bytes: leading_zero_bytes,
+            };
+            self.send_direct_message(peer_id, direct_message)?
+        }
         let elapsed = start.elapsed();
         trace!("{:?} created proof data in {}.{:06} seconds.  Min section size: {}, Target size: \
```

Comment:
Yes, it turns out sending is pretty much instantaneous (the calls, I mean: sending is asynchronous).
But this part will change again anyway, with Fraser's next PR.